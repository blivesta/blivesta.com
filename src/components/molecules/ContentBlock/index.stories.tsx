import React from 'react';
// import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import ContentBlock from './';

storiesOf(`molecules|ContentBlock`, module).add(`default`, () => (
  <ContentBlock id="foo" title="foo">
    content
  </ContentBlock>
));
