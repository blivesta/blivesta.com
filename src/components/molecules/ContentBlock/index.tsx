import * as React from 'react';
import styled, { css } from 'styled-components';

import Container from '../../atoms/Container';

const Wrapper = styled.section``;

const Flex = styled.div<ContentBlockProps>`
  display: flex;
  padding-bottom: 20vh;
  ${props =>
    props.logo &&
    css`
      padding-top: 15vh;
      padding-bottom: 30vh;
    `}

  ${props => props.theme.media.sm`
    padding-right: ${props.theme.spaces[8]};
  `}
`;

const TitleColumn = styled.div`
  width: 44px;
  margin-right: ${props => props.theme.spaces[3]};
  position: relative;

  ${props => props.theme.media.sm`
    width: 80px;
  `}
`;

const H1 = styled.h1`
  display: flex;
  position: absolute;
  top: 0;
  right: ${props => props.theme.spaces[4]};
  margin: 0;
  font-family: ${props => props.theme.fonts.en};
  font-size: ${props => props.theme.fontSizes[1]};
  color: ${props => props.theme.colors.baseText[1]};
  text-transform: uppercase;
  white-space: nowrap;
  letter-spacing: 0.15em;
  transform: rotate(-90deg);
  transform-origin: top right;

  ${props => props.theme.media.sm`
    font-size: ${props.theme.fontSizes[2]};
  `}
`;

const MainColumn = styled.div`
  flex: 1;
`;

interface ContentBlockProps {
  id?: string;
  title?: string;
  logo?: React.ReactNode;
  children: React.ReactNode;
}

const ContentBlock = ({ id, title, logo, children }: ContentBlockProps) => (
  <Wrapper id={id}>
    <Container>
      <Flex logo={logo}>
        <TitleColumn>{logo || <H1>{title}</H1>}</TitleColumn>
        <MainColumn>{children}</MainColumn>
      </Flex>
    </Container>
  </Wrapper>
);

export default ContentBlock;
