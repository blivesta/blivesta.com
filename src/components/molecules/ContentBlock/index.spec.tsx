import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ContentBlock from '.';

describe('ContentBlock', () => {
  it('snapshot', () => {
    const tree = shallow(<ContentBlock>...</ContentBlock>);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
