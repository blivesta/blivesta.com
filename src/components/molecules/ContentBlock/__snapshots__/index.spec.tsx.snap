// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBlock snapshot 1`] = `
<styled.section>
  <styled.div>
    <styled.div>
      <styled.div>
        <styled.h1 />
      </styled.div>
      <styled.div>
        ...
      </styled.div>
    </styled.div>
  </styled.div>
</styled.section>
`;
